### PR TITLE
Clean unused code in seasalt_model.F90

### PR DESCRIPTION
**Pull request by mingxuanwupnnl**
clean unused code in seasalt emission code.

@tangshuaiqi @brigaudet @singhbalwinder Do you think we should remove the code related to the local debug flag debug_mam_mom, which is always false. It is related to debug output. I tend to remove them, but I'm not so sure.

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 [hunk 1]
```diff
@@ -30,29 +30,9 @@ module seasalt_model
   public :: init_ocean_data           ! initialize ocean data variables
   public :: ocean_data_readnl         ! read ocean data namelist
 
-#if  ( defined MODAL_AERO_9MODE )
-  integer, parameter :: nslt = 4
-#else
   integer, parameter :: nslt = max(3,ntot_amode-3)
-#endif
   integer, parameter :: nnum = nslt
```
Comment (tangshuaiqi) [inline components/eam/src/chemistry/modal_aero/seasalt_model.F90:34]: can we merge them into one variable?

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 [hunk 2]
```diff
@@ -469,7 +408,6 @@ subroutine seasalt_emis(u10, u10cubed, lchnk, srf_temp, ocnfrc, ncol, cflx, emis
 
     fi(:ncol,:nsections) = fluxes( srf_temp, u10cubed, ncol )
 
-    calculate_organic_fraction: if ( has_mam_mom ) then
 
        nullify(chla)
```
Comment (tangshuaiqi) [inline components/eam/src/chemistry/modal_aero/seasalt_model.F90]: please change the indent as the if condition is removed. same as other places.

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 [hunk 3]
```diff
@@ -658,34 +515,19 @@ subroutine seasalt_emis(u10, u10cubed, lchnk, srf_temp, ocnfrc, ncol, cflx, emis
...
                       ! Mixing state 1: external mixture, add OM to mass and number
                       ! Mixing state 3: internal mixture, add OM to mass and number
                       cflx(:ncol,mn) = cflx(:ncol,mn) + cflx_help2(:ncol) * &
                                        (1._r8 / (1._r8 - om_ssa(:ncol, i)) - 1._r8)
-                   else
-                      ! Unknown mixing state assumption
-                      write(iulog, *) "Error: Unknown mixing_state value in seasalt_model.F90"
-                      exit
-                   end if
                 end if
```
Comment (tangshuaiqi) [inline components/eam/src/chemistry/modal_aero/seasalt_model.F90]: remove the spaces in `end if` and `end do`

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 [hunk 4]
```diff
@@ -697,14 +539,7 @@ subroutine seasalt_emis(u10, u10cubed, lchnk, srf_temp, ocnfrc, ncol, cflx, emis
...
-                   !                 total number not modified
-                   ! Mixing state 2: internal mixture, replace mass with OM,
-                   !                 total number not modified
-                   cflx(:ncol,mm) = cflx(:ncol,mm) + cflx_help2(:ncol) &
-                        * mass_frac_bub_section(:ncol, n, i)
-                else if ( ( mixing_state == 1 ) .or. ( mixing_state == 3 ) ) then
+             
                    ! Mixing state 1: external mixture, add OM to mass and number
                    ! Mixing state 3: internal mixture, add OM to mass and number
                    where (om_ssa(:ncol, i) .gt. 0.0_r8) ! avoid division by zero
```
Comment (tangshuaiqi) [inline components/eam/src/chemistry/modal_aero/seasalt_model.F90]: change .gt. to >

**Review by tangshuaiqi — APPROVED** [review]
Nice clean up. I made a few comments but notice that this is only the initial clean up so you can address them in the next PR. For the debug flag debug_mam_mom, as these debug output are not executed, I think removing them won't affect the BFB historical test (unlike my old case that the outfld lines are executed so removing them will fail historical file test).

**Review by brigaudet — APPROVED** [review]
Looks good!

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 (hunk 2 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/seasalt_model.F90]: changed the indent for subroutines.

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 (hunk 3 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/seasalt_model.F90]: changed them to endif and enddo

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 (hunk 4 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/seasalt_model.F90]: changed them to symbols

File: components/eam/src/chemistry/modal_aero/seasalt_model.F90 (hunk 1 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/modal_aero/seasalt_model.F90:34]: will check in the next PRs

**Comment by mingxuanwupnnl** [comment]
all tests passed.
  PASS ERP_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_Ly1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS.ne30pg2_EC30to60E2r2.F2010.compy_intel

**Pull request merged**